src/controllers/userProfileController.js

```diff
@@ -371,7 +373,18 @@ const userProfileController = function (UserProfile) {
         userData = allUserData[userIdx];
       }
       if (
-        await hasPermission(req.body.requestor, 'putUserProfileImportantInfo')
+
+        (await hasPermission(
+          req.body.requestor,
+          "putUserProfileImportantInfo"
+        )) ||
+        (await hasPermission(req.body.requestor, "manageAdminLinks"))
+      ) {
+        record.role = req.body.role;
+        record.isRehireable = req.body.isRehireable;
+        record.isActive = req.body.isActive;
+
+       
```

Comment:
```suggestion
        await hasPermission(req.body.requestor, "manageAdminLinks")
      ) {
        record.adminLinks = req.body.adminLinks;

       if (await hasPermission(req.body.requestor, "putUserProfileImportantInfo")
```
something more like this. The brackets from the if statement are causing syntax errors.